app/controllers/tasks_controller.rb

```diff
@@ -3,14 +3,31 @@ class TasksController < ApplicationController
   before_action :set_task, only: [:update, :destroy, :done]
 
   def index
-    @search = Task.search(params[:q])
-    @tasks = @search.result.on_user.where(is_done: false).sort_by(&:least_time_per)
+    @filter = {title: "タスク一覧", path: tasks_path}
+    @search = Task.on_user.where(is_done:false).search(params[:q])
+    @tasks = @search.result.sort_by(&:least_time_per)
   end
 
   # get
-  def doned
-    @search = Task.where(is_done: true).search(params[:q])
-    @tasks = @search.result.on_user.sort_by(&:least_time_per)
+  def completed
+    @filter = {title: "終了済みのタスク", path: completed_tasks_path}
+    @search = Task.on_user.where(is_done: true).search(params[:q])
+    @tasks = @search.result.sort_by(&:least_time_per)
+    render 'filter'
+  end
+
+  def today
+    @filter = {title: "今日のタスク", path: today_tasks_path}
+    @search = Task.on_user.where(deadline_at: Time.zone.today..Time.zone.tomorrow, is_done: false).search(params[:q])
+    @tasks = @search.result.sort_by(&:least_time_per)
+    render 'filter'
+  end
+
+  def weekly
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for weekly is too high. [15.3/15] [:ref](http://c2.com/cgi/wiki?AbcMetric)